packages/webview_flutter/webview_flutter_platform_interface/lib/src/v4/types/load_request_params.dart

```diff
@@ -0,0 +1,129 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:typed_data';
+
+import 'package:flutter/foundation.dart';
+import 'package:plugin_platform_interface/plugin_platform_interface.dart';
+import 'package:webview_flutter_platform_interface/src/v4/webview_controller_delegate.dart';
+
+import '../webview_platform.dart';
+
+/// Defines the supported HTTP methods for loading a page in
+/// [WebViewControllerDelegate].
+enum LoadRequestMethod {
+  /// HTTP GET method.
+  get,
+
+  /// HTTP POST method.
+  post,
+}
+
+/// Extension methods on the [LoadRequestMethod] enum.
+extension LoadRequestMethodExtensions on LoadRequestMethod {
+  /// Converts [LoadRequestMethod] to [String] format.
+  String serialize() {
+    switch (this) {
+      case LoadRequestMethod.get:
+        return 'get';
+      case LoadRequestMethod.post:
+        return 'post';
+    }
+  }
+}
+
+/// Defines the parameters that can be used to load a page with the
+/// [WebViewControllerDelegate].
+///
+/// Platform specific implementations can add additional fields by extending this
+/// class and provide a factory method that takes the
+/// [LoadRequestParamsDelegate] as a parameter.
+///
+/// {@tool sample}
+/// This example demonstrates how to extend the [LoadRequestParamsDelegate] to
+/// provide additional platform specific parameters.
+///
+/// Note that the additional parameters should always accept `null` or have a
+/// default value to prevent breaking changes.
+///
+/// ```dart
+/// class AndroidLoadRequestParamsDelegate extends LoadRequestParamsDelegate {
+///   AndroidLoadRequestParamsDelegate._(
+///     LoadRequestParamsDelegate loadRequestParams,
+///     this.historyUrl,
+///   ) : super(
+///     uri: loadRequestParams.uri,
+///     method: loadRequestParams.method,
+///     headers: loadRequestParams.headers,
+///     body: loadRequestParams.body,
+///   );
+///
+///   factory AndroidLoadRequestParamsDelegate.fromLoadRequestParamsDelegate(
+///     LoadRequestParamsDelegate loadRequestParams, {
+///     Uri? historyUrl,
+///   }) {
+///     return AndroidLoadRequestParamsDelegate._(
+///       loadRequestParams: loadRequestParams,
+///       historyUrl: historyUrl,
+///     );
+///   }
+///
+///   final Uri? historyUrl;
+/// }
+/// ```
+/// {@end-tool}
+class LoadRequestParamsDelegate extends PlatformInterface {
+  /// Creates a new [LoadRequestParamsDelegate].
+  factory LoadRequestParamsDelegate({
+    required Uri uri,
+    required LoadRequestMethod method,
+    required Map<String, String> headers,
```

Comment:
I think this can have a default value of `const <String, String>{}` and not be required.